Lib/fontParts/base/base.py

```diff
@@ -95,108 +89,177 @@ def __get__(self, obj, cls):
-        """
-        Subclasses may override this method.
+    def _init(self, *args: Any, **kwargs: Any) -> None:
+        r"""Initialize the native object.
+
+        This is the environment implementation of :meth:`BaseObject.__init__`.
+
+        :param \*args: Any positional arguments.
+        :param \**kwargs: Any keyword arguments.
+
+        .. note::
+
+            Subclasses may override this method.
+
         """
-        pass
 
     # ----
     # repr
     # ----
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         contents = self._reprContents()
         if contents:
-            contents = " ".join(contents)
-            contents = " " + contents
+            contentString = " ".join(contents)
+            contentString = " " + contentString
         else:
-            contents = ""
-        s = "<{className}{contents} at {address}>".format(
-            className=self.__class__.__name__,
-            contents=contents,
-            address=id(self)
-        )
-        return s
+            contentString = ""
+        return f"<{self.__class__.__name__}{contentString} at {id(self)}>"
 
     @classmethod
-    def _reprContents(cls):
-        """
-        Subclasses may override this method to
-        provide a list of strings for inclusion
-        in ``__repr__``. If so, they should call
-        ``super`` and append their additions
-        to the returned ``list``.
+    def _reprContents(cls) -> List[str]:
+        """Provide a list of strings for inclusion in :meth:`BaseObject.__repr__.
+
+        :return: A :class:`list` of :class:`str` items.
+
+        .. note::
+
+            Subclasses may override this method.
+            If so, they should call :class:`super` and append their additions.
+
         """
         return []
 
     # --------
     # equality
     # --------
 
-    def __eq__(self, other):
-        """
-        Subclasses may override this method.
+    def __eq__(self, other: Any) -> bool:
+        """Check for equality with another object.
+
+        :param other: The object to compare with.
+        :return: :obj:`True` if the objects are equal, :obj:`False` otherwise.
+
+        .. note::
+
+            Subclasses may override this method.
+
         """
         if isinstance(other, self.__class__):
             return self.naked() is other.naked()
         return NotImplemented
 
-    def __ne__(self, other):
-        """
-        Subclasses must not override this method.
+    def __ne__(self, other: Any) -> bool:
+        """Check for inequality with another object.
+
+        :param other: The object to compare with.
+        :return: :obj:`True` if the objects are unequal, :obj:`False` otherwise.
+
+        .. note::
+
+            Subclasses must not override this method.
+
         """
         equal = self.__eq__(other)
-        return NotImplemented if equal is NotImplemented else not equal
+        return False if equal is NotImplemented else not equal
```

Comment:
Sorry, this was a solution to `mypy` throwing an error since the return type is only expected to be a `bool`. It was not correct, though. Simply refactoring the return logic into a multi-liner solved it.